### PR TITLE
feat(panel): Update the panel position on scroll.

### LINKED ISSUE #7897: mdPanel: Reposition on scroll.

### PR DESCRIPTION
**Pull request by ErinCoughlan**
Fixes #7897

@KarenParker @gmoothart @DerekLouie @ThomasBurleson @jelbourn Please review.

### CODE REVIEW

**Comment by KarenParker** [comment]
👍

**Comment by DerekLouie** [comment]
LGTM

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -1288,6 +1305,31 @@ MdPanelRef.prototype._configureClickOutsideToClose = function() {
 
 
 /**
+ * Configures the listeners for updating the panel position on scroll.
+ * @private
+*/
+MdPanelRef.prototype._configureScrollListener = function() {
+  var updatePosition = angular.bind(this, this._updatePosition);
+  var debouncedUpdatePosition = this._$$rAF.throttle(updatePosition);
```
Comment (jelbourn) [inline src/components/panel/panel.js]: What's the timer on this? And is `throttle` actually performing a debounce?

Comment (ErinCoughlan) [inline src/components/panel/panel.js]: `throttle` is doing the debounce, once per animation frame. It uses the last event that happened before the frame.

Comment (ThomasBurleson) [inline src/components/panel/panel.js]: Yes - we forget to rename it. (sigh)

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -1311,6 +1328,31 @@ MdPanelRef.prototype._configureClickOutsideToClose = function() {
...
+      debouncedUpdatePosition();
+    }
+  };
+
+  // Add listeners.
+  this._$window.addEventListener('scroll', onScroll, true);
+
+  // Queue remove listeners function.
+  this._removeListeners.push(function() {
+    self._$window.removeEventListener('scroll', onScroll);
```
Comment (ThomasBurleson) [inline src/components/panel/panel.js]: I think you are missing an argument. If you use the *capture* phase in the `addEventListener( )`, then you must match when removing:

```js
self._$window.removeEventListener('scroll', onScroll, true);
```

**Comment by ErinCoughlan** [comment]
Fixed the removeListener. PTAL.

**Comment by ErinCoughlan** [comment]
LGTM from @ThomasBurleson via Slack.

**Pull request merged**